tests/integration/relationships/inverse-relationships-test.js

```diff
@@ -463,8 +463,8 @@ testInDebug("Inverse relationships that don't exist throw a nice error for a bel
   }, /We found no inverse relationships by the name of 'testPost' on the 'user' model/);
 });
 
-test("inverseFor is only called when inverse is not null", function(assert) {
-  assert.expect(2);
+test("inverseFor short-circuits when inverse is null", function(assert) {
```

Comment:
Looking more closely, we don't need to introduce `_inverseFor`; the point was just to separate the API from the work so we can test that the expensive path is not taken, but this separation already exists between `inverseFor` and `_findInverseFor`, so I've updated the tests to assert that `inverseFor` short-circuits `_findInverseFor` rather than `_inverseFor`.